### PR TITLE
Challenge21: golang binary! (and some overdue bash and java cleanups)

### PR DESCRIPTION
**Pull request by commjoen**
Thank you for submitting a pull request to the WrongSecrets app!

### CODE REVIEW

**Review by bendehaan — APPROVED** [review]

File: src/main/resources/explanations/challenge21.adoc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+=== Obfuscating part 3: the Golang binary
+
+Our third language of choice for a compiled application is Golang. With the rise of its popularity, we see an increase of secrets hidden inside the binaries. Can you find the secret in our binary?
+
+Let's debunk the "secrets are hard to find in native compiled applications" myth for C++: can you find the secret in https://github.com/commjoen/wrongsecrets/tree/master/src/main/resources/executables/wrongsecrets-golang[wrongsecrets-golang] (or https://github.com/commjoen/wrongsecrets/tree/master/src/main/resources/executables/wrongsecrets-golang-arm[wrongsecrets-golang-arm], https://github.com/commjoen/wrongsecrets/tree/master/src/main/resources/executables/wrongsecrets-golang-linux[wrongsecrets-golang-linux])?
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge21.adoc:5]:
```suggestion
=== Obfuscating part 3: the Go binary

Our third language of choice for a compiled application is Go. With the rise of its popularity, we see an increase of secrets hidden inside the binaries. Can you find the secret in our binary?

Let's debunk the "secrets are hard to find in native compiled applications" myth for Go: can you find the secret in https://github.com/commjoen/wrongsecrets/tree/master/src/main/resources/executables/wrongsecrets-golang[wrongsecrets-golang] (or https://github.com/commjoen/wrongsecrets/tree/master/src/main/resources/executables/wrongsecrets-golang-arm[wrongsecrets-golang-arm], https://github.com/commjoen/wrongsecrets/tree/master/src/main/resources/executables/wrongsecrets-golang-linux[wrongsecrets-golang-linux])?
```

File: src/main/resources/explanations/challenge21_hint.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+This challenge is specifically looking at a secret in a Golang binary
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge21_hint.adoc:1]:
```suggestion
This challenge is specifically looking at a secret in a Go binary
```

File: src/main/resources/explanations/challenge21_hint.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+This one is a little harder, as we used Cobra to create the CLI introducing some more overhead.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge21_hint.adoc:3]:
```suggestion
This one is a little harder, as we used Cobra to create the CLI, introducing some more overhead.
```

File: src/main/resources/explanations/challenge21_hint.adoc [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+You can solve this challenge using the following steps:
+
+1. Find the secrets with https://ghidra-sre.org/[Ghidra].
+- Install https://ghidra-sre.org/[Ghidra].
+- Start it whit `ghidraRun`.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge21_hint.adoc:8]:
```suggestion
- Start it with `ghidraRun`.
```

File: src/main/resources/explanations/challenge21_hint.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+- Load the application `wrongsecrets-golang` into ghidra by choosing a new project, then import the file and then doubleclick on it.
+- Allow the Ghidra to analyze the application. Note that this takes much longer as our binary is a lot larger.
+- Go to the data type manager in the bottom left, now filter for `string`, now right-click at `string` as a member of `wrongsecrets-golang` and select `find uses of`.
+- Now filter for known keywords: you should easily be able to find the secret now!
+
+2. Find the secrets with https://www.radare.org[radare2].
+- Install https://www.radare.org[radare2] with either `brew install radare2` on Mac or follow these steps: `git clone https://github.com/radareorg/radare2; cd radare2 ; sys/install.sh`
+- Launch r2 analysis with `$ r2 -A wrongsecrets-golang`
+- Start a search for the string with `/w secret`
+- Now take the results and look for possible answers, how about `/w his is the secret in Golgang` ? You should be able to find the secret now.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge21_hint.adoc:18]:
```suggestion
- Now take the results and look for possible answers, how about `/w his is the secret in Golang` ? You should be able to find the secret now.
```

**Comment by commjoen** [comment]
First version is live at https://arcane-scrubland-42646.herokuapp.com/ . We will merge this together with the 1st of the month incoming patches as we will have to test it in the different clouds again...

**Pull request merged**